.travis.yml

```diff
@@ -125,105 +144,95 @@ matrix:
       after_failure:
         - build-support/bin/ci-failure.sh
 
-    # Deploy Pex
-    - os: linux
+    # Build and deploy an OSX / Linux Pants pex to S3.
+    - <<: *s3_deploy_config
+      stage: Build Pants Pex
+      os: linux
       language: python
-      stage: Deploy Pants Pex
-      env:
-        - PANTS_PEX_RELEASE=stable
       script:
         - ./build-support/bin/release.sh -p
-      deploy:
-        # See https://docs.travis-ci.com/user/deployment/releases/
-        provider: releases
-        # The pantsbuild-ci-bot OAuth token, see the pantsbuild vault for details.
-        api_key:
-          secure: "u0aCsiuVGOg28YxG0sQUovuUm29kKwQfFgHbNz2TT5L+cGoHxGl4aoVOCtuwWYEtbNGmYc8/3WRS3C/jOiqQj6JEgHUzWOsnfKUObEqNhisAmXbzBbKc0wPQTL8WNK+DKFh32sD3yPYcw+a5PTLO56+o7rqlI25LK7A17WesHC4="
-        file_glob: true
-        file: dist/deploy/pex/*
-        skip_cleanup: true
-        on:
-          # We only release a pex for Pants releases, which are tagged.
-          tags: true
-          repo: pantsbuild/pants
+      before_deploy:
+        # Wheels were downloaded to dist/deploy from S3 and we don't to re-upload them to S3, but
+        # we do want to upload the pex so shuffle dist/deploy/ contents pre-deploy.
+        - "rm -rf dist/deploy && mkdir -p dist/deploy && mv dist/pants.*.pex dist/deploy/"
 
     - <<: *default_test_config
       env:
         - SHARD="Various pants self checks and lint"
       script:
-        - ./build-support/bin/ci.sh -x -cejlpn 'Various pants self checks and lint'
+        - "CI_USE_PREBUILT_PANTS_PEX=1 ./build-support/bin/ci.sh -x -cejlpn"
 
     - <<: *default_test_config
       env:
         - SHARD="Unit tests for pants and pants-plugins - shard 1"
       script:
-        - ./build-support/bin/ci.sh -x -efkmrcnt -u 0/2 "${SHARD}"
+        - "CI_USE_PREBUILT_PANTS_PEX=1 ./build-support/bin/ci.sh -x -efkmrcnt -u 0/2"
 
     - <<: *default_test_config
       env:
         - SHARD="Unit tests for pants and pants-plugins - shard 2"
       script:
-        - ./build-support/bin/ci.sh -x -efkmrcnt -u 1/2 "${SHARD}"
+        - "CI_USE_PREBUILT_PANTS_PEX=1 ./build-support/bin/ci.sh -x -efkmrcnt -u 1/2"
 
     - <<: *default_test_config
       env:
         - SHARD="Python contrib tests - shard 1"
       script:
-        - ./build-support/bin/ci.sh -x -efkmrcjlpt -y 0/2 "${SHARD}"
+        - "CI_USE_PREBUILT_PANTS_PEX=1 ./build-support/bin/ci.sh -x -efkmrcjlpt -y 0/2"
```

Comment:
It can't unless there is yaml-foo I'm missing. Inheritance is only 1 level deep so defining envs in an inheritor wipes the inherited. I could complexify things with `default_test_config` inheriting a `default_env` and then overriding that nested in all these, but that seems like not-a-win.